lxd/instance/drivers/driver_qemu_templates.go

```diff
@@ -9,11 +11,27 @@ import (
 	"github.com/canonical/lxd/shared/osarch"
 )
 
-// qemuHostDriveDeviceID returns the device ID to use for a host drive share.
-func qemuHostDriveDeviceID(deviceName string, protocol string) string {
-	suffix := "-" + protocol
-	maxNameLength := qemuDeviceIDMaxLength - (len(qemuDeviceIDPrefix) + len(suffix))
-	return fmt.Sprintf("%s%s%s", qemuDeviceIDPrefix, hashIfLonger(filesystem.PathNameEncode(deviceName), maxNameLength), suffix)
+// qemuDeviceNameOrID generates a QEMU device name or ID.
+// Respects the property length limit by hashing the device name when necessary. Also escapes / to -, and - to --.
+func qemuDeviceNameOrID(prefix string, deviceName string, suffix string, maxLenght int) string {
+	baseName := filesystem.PathNameEncode(deviceName)
+	maxNameLength := maxLenght - (len(prefix) + len(suffix))
```

Comment:
typo in length var name, also picked up by tests, do do please check those results to pre-empt the review. Thanks